samtranslator/model/apigateway.py

```diff
@@ -39,7 +40,7 @@ class ApiGatewayStage(Resource):
             'DeploymentId': PropertyType(True, is_str()),
             'Description': PropertyType(False, is_str()),
             'RestApiId': PropertyType(True, is_str()),
-            'StageName': PropertyType(True, one_of(is_str(), is_type(dict))),
+            'StageName': PropertyType(False, one_of(is_str(), is_type(dict))),
```

Comment:
Why did you change this to `False`?

Reply:
I followed Sanath's previous change for stage stage fix. If this is required I will have to set it, which is not what we want.

Reply:
I think (but am not sure) that we still want this StageName property set, but not the other one that is on the `Deployment` resource. Either way, this will still work (we're still setting the StageName property on the Stage resource and conditionally removing it from the Deployment resource), so this is more of a technicality than a requirement.

Reply:
I'd love to make this optional. Is there any reason not to?

Reply:
If we decide to make this optional, we need to decide on a default value ("prod"?) and update docs.

Reply:
I think that should be a part of another cr. I don't want to stray too far from Sanath's changes for this bug previously.

Reply:
I removed this change thanks Keeton!